pkg/provisioner/ironic/validatemanagementaccess_test.go

```diff
@@ -179,3 +181,168 @@ func TestValidateManagementAccessNewCredentials(t *testing.T) {
 	newValues := updates[0].Value.(map[string]interface{})
 	assert.Equal(t, "test.bmc", newValues["test_address"])
 }
+
+func TestValidateManagementAccessLinkExistingIronicNodeByMAC(t *testing.T) {
+	// Create an Ironic node, and then create a host with a matching MAC
+	// Test to see if the node was found, and if the link is made
+
+	existingNode := nodes.Node{
+		UUID: "33ce8659-7400-4c68-9535-d10766f07a58",
+	}
+
+	existingNodePort := ports.Port{
+		NodeUUID: existingNode.UUID,
+		Address:  "11:11:11:11:11:11",
+	}
+
+	createCallback := func(node nodes.Node) {
```

Comment:
We see this pattern a few times. Maybe we can add a FailCreateNode() method to the IronicMock in another PR.